### PR TITLE
Updated CHANGELOG.md in preparation for the next version release

### PR DESCRIPTION
**Pull request by ttmc**
@r-marques @vrde @diminator @rhsimplex Please take a quick look and let me know if I missed any _significant_ merged pull requests since the version 0.3.0 release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,19 +12,50 @@ For reference, the possible headings are:
...
+## [Unreleased] - YYYY-MM-DD
+Tag name: 
+= commit: 
+committed: 
+
+### Added
+- New `Dockerfile-dev` Docker file to make it easier for developers to _develop_ BigchainDB with Docker. One can run all unit tests with Docker again. [Pull Request #313](https://github.com/bigchaindb/bigchaindb/pull/313)
+- Transactions in invalid blocks are copied to the backlog: [Pull Request #221](https://github.com/bigchaindb/bigchaindb/pull/221).
+- New `bigchaindb` command to set the number of RethinkDB shards (in both tables): [Pull Request #258](https://github.com/bigchaindb/bigchaindb/pull/258)
+- New command to gather some metrics from a cluster: [Pull Request #266](https://github.com/bigchaindb/bigchaindb/pull/266). Note: This method has some issues so we advise against using it for now.
```
Comment (r-marques) [inline CHANGELOG.md]: I guess we should omit this until we are ready to ship benchmarks

**Comment by r-marques** [comment]
there is also #271 speeds up the instantiation of `Bigchain()` greatly improving the overall performance, and #279 better handling of outdated setuptools, there were several issues related to this.

**Comment by ttmc** [comment]
Okay @r-marques I will add those and push them to this branch.

@vrde would like the next version to use the secondary index on get_transaction, using PR #324 from @rhsimplex (rather than PR #320 ). So we'll wait for that to get sorted and merged.

Schema validation (PR #311 ) isn't as critical and can wait until the next version.

File: CHANGELOG.md (hunk 1 above)
Comment (ttmc) [inline CHANGELOG.md]: Okay, I will omit mention of PR #266

**Comment by ttmc** [comment]
The live CHANGELOG now includes everything up to the escrow PR #329 so we can do a next release.

**Comment by diminator** [comment]
Ok, let me know when you want to release!
I can help with setting up the pypi image

**Pull request merged**